docs/compilers/CSharp/Compiler Breaking Changes - post VS2019.md

```diff
@@ -8,3 +8,6 @@ Each entry should include a short description of the break, followed by either a
     bool M<T>(T t) => t is null;
     ```
 However, in *Visual Studio 2019* we improperly permitted this to compile in language versions `7.0`, `7.1`, `7.2`, and `7.3`.  In *Visual Studio 2019 Update 1* we will make it an error (as it was in *Visual Studio 2017*), and suggest updating to `preview` or `8.0`.
+
+2. https://github.com/dotnet/roslyn/issues/38226 When there exists a common type among those arms of a switch expression that have a type, but there are some arms that have an expression without a type (e.g. `null`) that cannot convert to that common type, the compiler improperly inferred that common type as the natural type of the switch expression. That would cause an error.  In VS 2019 Update 4, we fixed the compiler to no longer consider such a switch expression to have a common type.  This may permit some programs to compile without error that would produce an error in the previous version.
```

Comment:
If you build an app built with the 3.0 SDK in 16.4, which compiler is used during build time? One that will have this fix or one that won't? My worry is if the langver isn't changing, does the IDE know which rules to apply when determining if casts can be removed?